### ISSUE TITLE
Is Inverse Dynamics supported?

### ISSUE THREAD

**Issue by mklvcm** [AndresTraks/BulletSharpPInvoke#67]
Hi Andres,

I'm looking for a library that implements Inverse Dynamics (calculating joint torques) for a robot simulation, and I'd like it in C#. 
I don't care too much about rendering (only for diagnostics).
I know that original Bullet supports Inverse Dynamics, but does BulletSharpPinvoke or BulletSharp?
I can't see anything like that in code or demos.
Also, do you provide support for reading URDF files?

It not supported, do you know of any C# implementations?

Thanks,
Mike.

**Comment by AndresTraks** [comment]
Hi,
Inverse Dynamics is not yet implemented. but it would be cool to have that.

I did start with writing a URDF loader:
https://github.com/AndresTraks/BulletSharpPInvoke/tree/master/BulletSharp/demos/DemoFramework/FileLoaders
https://github.com/AndresTraks/BulletSharpPInvoke/tree/master/BulletSharp/demos/UrdfDemo
but it doesn't work properly. The transforms are messed up and I found it hard to follow the URDF specification to try to figure out how it's supposed to work.
Plus, Bullet has its own modifications to the file format.

I'll add this back on my todo list, but can't promise anything.

**Comment by mklvcm** [comment]
Thanks Andres!

**Comment by Traveller-Jason** [comment]
Hi Andres，I'm looking for a library that implements Inverse Kinematics (calculating joint angles) for a robot simulation, and I'd like it in C#. Also, do you provide support for reading URDF files? I have a similar problem with mklvcm, and I am curious if there are any new developments or suggestions at the moment. @AndresTraks
